php-checks/src/main/java/org/sonar/php/checks/phpunit/AssertionArgumentOrderCheck.java

```diff
@@ -52,23 +57,47 @@ public void visitFunctionCall(FunctionCallTree tree) {
     if (arguments.size() >= 2 && assertion.isPresent() && assertion.get().hasExpectedValue() && !hasNamedArgument(tree)) {
       ExpressionTree expected = arguments.get(0).value();
       ExpressionTree actual = arguments.get(1).value();
-      if (isLiteralOrClassName(actual) && !isLiteralOrClassName(expected)) {
+      if (isLiteralOrClassNameOrParameter(actual) && !isLiteralOrClassNameOrParameter(expected)) {
         newIssue(actual, MESSAGE).secondary(expected, SECONDARY_MESSAGE);
       }
     }
 
     super.visitFunctionCall(tree);
   }
 
-  private static boolean isLiteralOrClassName(ExpressionTree expression) {
+  private static boolean isLiteralOrClassNameOrParameter(ExpressionTree expression) {
     return assignedValue(expression).is(LITERAL) ||
-      (expression instanceof MemberAccessTree &&
-        isStaticAccessWithName((MemberAccessTree) expression, "class"));
+      isStaticAccessWithName(expression, "class") ||
+      isDefinedAsParameter(expression);
```

Comment:
I believe the description of the ticket is not correct: `assertSame` has the same order as other assertions, we need to support parameters as expected values for all assertions. Should it be reflected in the ticket also?

Reply:
Yes, the description is wrong, I was confused at the beginning. I will fix it.